recipe-server/client/control/tests/components/recipes/test_ImportExperimentsPage.js

```diff
@@ -0,0 +1,206 @@
+  });
+  it('should contain ExperimentRejectModal', () => {
+    const wrapper = shallow(<ImportExperimentsPage />);
+    expect(wrapper.contains(<ExperimentRejectModal />)).toBeTruthy();
+  });
+  it('should contain ExperimentsList', () => {
+    const wrapper = shallow(<ImportExperimentsPage />);
+    expect(wrapper.contains(<ExperimentsList />)).toBeTruthy();
+  });
+});
+
+describe('<ExperimentsList>', () => {
+  const props = {
+    experiments: ExperimentFactory.buildMany(3),
+  };
+
+  it('should work', () => {
+    const wrapper = () => shallow(<ExperimentsList.WrappedComponent {...props} />);
+    expect(wrapper).not.toThrow();
+  });
+  it('should contain Experiments', () => {
+    const wrapper = shallow(<ExperimentsList.WrappedComponent {...props} />);
+    expect(wrapper.find(Experiment).length).toEqual(props.experiments.length);
+  });
+});
+
+describe('<ExperimentRejectModal>', () => {
+  const props = {
+    rejectExperiment: () => {},
+    rejectedExperiment: null,
+    selectRejectedExperiment: () => {},
+  };
+
+  it('should work', () => {
+    const wrapper = () => shallow(<ExperimentRejectModal.WrappedComponent {...props} />);
+    expect(wrapper).not.toThrow();
+  });
+  it('should not contain a Modal if no rejectedExperiment is set', () => {
+    const wrapper = shallow(<ExperimentRejectModal.WrappedComponent {...props} />);
+    expect(wrapper.find(Modal).length).toEqual(0);
+  });
+  it('should contain a Modal if rejectedExperiment is set', () => {
+    const testProps = {
+      rejectExperiment: () => {},
+      rejectedExperiment: ExperimentFactory.build(),
+      selectRejectedExperiment: () => {},
+    };
+
+    const wrapper = shallow(<ExperimentRejectModal.WrappedComponent {...testProps} />);
+    expect(wrapper.find(Modal).length).toEqual(1);
+  });
+  it('should store the rejection reason in its state when edited', () => {
+    const reason = 'rejected for reasons.';
+    const wrapper = shallow(<ExperimentRejectModal.WrappedComponent {...props} />);
+    wrapper.instance().handleReasonChange({ target: { value: reason } });
+    expect(wrapper.state().reason).toEqual(reason);
+  });
+  it('should selectRejectedExperiment with null when canceled', () => {
+    const testProps = {
+      rejectExperiment: () => {},
+      rejectedExperiment: null,
+      selectRejectedExperiment: newRejectedExperiment => {
+        expect(newRejectedExperiment).toEqual(null);
+      },
+    };
+
+    const wrapper = shallow(<ExperimentRejectModal.WrappedComponent {...testProps} />);
+    wrapper.instance().handleCancel();
+  });
+  it('should call rejectExperiment with a reason when rejected', () => {
+    const reason = 'rejected for reasons.';
+    const experiment = ExperimentFactory.build();
+
+    const testProps = {
+      rejectExperiment: (rejectedExperiment, rejectionReason) => {
+        expect(rejectedExperiment.slug).toEqual(experiment.slug);
+        expect(rejectionReason).toEqual(reason);
```

Comment:
This type of call-assertion is a little less reliable, because this would also pass if the function is never called. This would probably be another good place to use a `jasmine.createSpy` and `expect(...).toHaveBeenCalledWith`.